dicthash/dicthash.py

```diff
@@ -66,11 +70,11 @@ def _generate_string_from_dict(d, blacklist, whitelist, prefix=''):
     """
     if whitelist is None:
         whitelist = d.keys()
-
     if blacklist is not None:
         whitelist = [key for key in whitelist if key not in blacklist]
-
-    raw = [_unpack_value(d[key], prefix + unicode(key)) for key in sorted(whitelist)]
+    # Sort whitelist according to the keys converted to str
+    whitelist_sorted = [elem[1] for elem in sorted(zip(map(str, whitelist), whitelist))]
```

Comment:
Yes, you're right, `sorted(whitelist, key=str)` works perfectly, thanks for suggesting. I thought I had tested that... 
Anyway, you need this because sorting variables of different types does not work  in Python3. For instance, this
`l = ['a', 'b', 1, 2]`
`sorted(l)`
throws the error
`TypeError: unorderable types: int() < str()`